FirebaseFunctionsSwift/Tests/IntegrationTests.swift

```diff
@@ -0,0 +1,652 @@
+        "FCMTokenTest",
+        requestAs: [String: Int].self,
+        responseAs: [String: Int].self
+      )
+
+      let data = try await function.call([:])
+      XCTAssertEqual(data, [:])
+    }
+  #endif
+
+  func testNull() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "nullTest",
+      requestAs: Int?.self,
+      responseAs: Int?.self
+    )
+    function.call(nil) { result in
+      do {
+        let data = try result.get()
+        XCTAssertEqual(data, nil)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 5)
+  }
+
+  #if compiler(>=5.5) && canImport(_Concurrency)
+    @available(iOS 15, tvOS 15, macOS 12, watchOS 8, *)
+    func testNullAsync() async throws {
+      let function = functions.httpsCallable(
+        "nullTest",
+        requestAs: Int?.self,
+        responseAs: Int?.self
+      )
+
+      let data = try await function.call(nil)
+      XCTAssertEqual(data, nil)
+    }
+  #endif
+
+  // No parameters to call should be the same as passing nil.
+  // If no parameters are required, then the non-typed API
+  // is more appropriate since it specifically avoids defining
+  // type.
+  //  func testParameterless() {
+  //  }
+  //
+  //
+  func testMissingResult() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "missingResultTest",
+      requestAs: Int?.self,
+      responseAs: Int?.self
+    )
+    function.call(nil) { result in
+      do {
+        _ = try result.get()
+      } catch {
+        let error = error as NSError
+        XCTAssertEqual(FunctionsErrorCode.internal.rawValue, error.code)
+        XCTAssertEqual("Response is missing data field.", error.localizedDescription)
+        expectation.fulfill()
+      }
+    }
+    waitForExpectations(timeout: 5)
+  }
+
+  #if compiler(>=5.5) && canImport(_Concurrency)
+    @available(iOS 15, tvOS 15, macOS 12, watchOS 8, *)
+    func testMissingResultAsync() async {
+      let function = functions.httpsCallable(
+        "missingResultTest",
+        requestAs: Int?.self,
+        responseAs: Int?.self
+      )
+      do {
+        _ = try await function.call(nil)
+        XCTFail("Failed to throw error for missing result")
+      } catch {
+        let error = error as NSError
+        XCTAssertEqual(FunctionsErrorCode.internal.rawValue, error.code)
+        XCTAssertEqual("Response is missing data field.", error.localizedDescription)
+      }
+    }
+  #endif
+
+  func testUnhandledError() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "unhandledErrorTest",
+      requestAs: [Int].self,
+      responseAs: Int.self
+    )
+    function.call([]) { result in
+      do {
+        _ = try result.get()
```

Comment:
Same here re: `XCTFail("Failed to throw error for missing result")`

Reply:
Yep. Good catch! Added here and below.